src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -362,12 +402,20 @@ def create_self_signed_cert(self):
         cert.sign(key, 'SHA256')
         return cert, key
 
+    @private
+    def parse_cert_date_string(self, date_value):
+        t1 = dateutil.parser.parse(date_value)
+        t2 = t1.astimezone(dateutil.tz.tzlocal())
+        return t2.ctime()
+
     @private
     @accepts(
-        Str('certificate', required=True)
+        Str('certificate', required=True, null=True)
```

Comment:
Whats the point of call load_certificate if its null?

Reply:
It is to handle cases where it is possible we might have an old malformed CA perhaps with it's value set to None. So when we parse the cert chain, we will get an exception for null values not allowed

Reply:
You should avoid doing that call if the value is None.